### DISCUSSION TITLE
Error generating thumbnail PursaSlicer MacOS

### DISCUSSION THREAD

**Discussion by jvcjbl** [mriscoc/Ender3V2S1#108] [Q&A] [answered]
<img width="853" alt="Screen Shot 2022-04-01 at 9 59 39 PM" src="https://user-images.githubusercontent.com/102492061/161372407-332c3def-5b7f-4c0c-87ab-59a4b7fdebf6.png">

This is the error I'm facing when I'm trying to generate a thumbnail on MacOS PursaSlicer. I was facing a permission error but I figured that out thanks to a member on the Facebook Group. I'm hoping someone on here can help me figure this out as it's the only issue I'm facing with PrusaSlicer thus far. Cheers!

**Comment by mriscoc** [comment, marked as answer]
Check if you have installed Python 3.x version, usually Pillow library is installed by default in Python 3.x if not, install Pillow library: 
```sh
pip install Pillow
```
with proper permissions. Update your Python version.

https://www.geeksforgeeks.org/how-to-install-pillow-on-macos/

**Comment by jvcjbl** [comment]
This rectified the issue with that. Maybe I'd suggest adding a note to the in the WiKi for MAC as I'm probably not the only one with this issue.

**Reply by mriscoc** [reply]
This could be enough?
https://github.com/mriscoc/Ender3V2S1/wiki/How-to-generate-a-gcode-preview#notes-for-mac-users-and-prusasuper-slicer

**Reply by jvcjbl** [reply]
That is excellent and would have helped greatly at first.

**Reply by leydonhd** [reply]
I have a problem with Prusa Slicer......when i try to open g-code with prusa g-code viewer....when i generate thumbail with slicer was no error..only when i try to open g-code file with viewer
![prusa](https://user-images.githubusercontent.com/46671340/161529440-cda9edad-bcef-49e1-8280-94648918ce18.png)

**Reply by mriscoc** [reply]
Add at the end of you G-code the line:
```gcode
; prusaslicer_config = end
```
Some versions don't close the G-code with that line.

**Reply by jvcjbl** [reply]
Any idea why it worked great for a few prints and then just stopped working after I added the Prusa SLA printer? I'm not saying you are wrong, but wouldn't me deleting prusaslicer and reinstalling it fix the issue? I reverted back to Cura for now. It's working wonderfully.

**Reply by mriscoc** [reply]
Only checking a generated G-code could be possible analyze it to verify what is wrong.

**Reply by jvcjbl** [reply]
I'm headed to bed. I'll slice something tomorrow and post it. Im almost tempted to leave well enough alone as good as Cura is working now

**Reply by jvcjbl** [reply]
[ezyzip.zip](https://github.com/mriscoc/Ender3V2S1/files/8413530/ezyzip.zip)
Here is just something random I sliced in PrusaSlicer with the thumbnail scripts. I think I'm going to have to ask in the other forum as this is the script for generating thumbnails and print previews which your firmware doesn't support. Maybe you still have an idea what's going on?

**Reply by mriscoc** [reply]
This firmware supports only a preview thumbnail of 230x180

**Reply by jvcjbl** [reply]
I'm aware... that's why I said that I may have to ask in that group as theirs shows a 50x50 thumbnail and a 180x180 preview. It was working flawlessly and then stopped when I added the SL1 as I've said. No clue why it won't work anymore when it was working perfectly. I couldn't get it to work with your firmware which is what I posed about initially but I fixed it. I switched back to their firmware as it seems to work more reliably with my printer for whatever reason and was having similar issues with PursaSlicer and my printer. Thanks for your time

**Reply by leydonhd** [reply]
> 

i have done like you said.....the problem persist...tell me the same message as in photo

**Reply by mriscoc** [reply]
You have a different issue, but I think that it is not firmware related, maybe it could be better to ask for help at the Prusa forum.
